drivers/flash/soc_flash_nrf5.c

```diff
@@ -18,8 +18,9 @@
 #if defined(CONFIG_SOC_FLASH_NRF5_RADIO_SYNC)
 #include <misc/__assert.h>
 #include <bluetooth/hci.h>
-#include "controller/ticker/ticker.h"
-#include "controller/include/ll.h"
+#include "bluetooth/controller/hal/ticker.h"
+#include "bluetooth/controller/ticker/ticker.h"
```

Comment:
I understand the point that the ticker module is not (supposed to be) public, however: 

would it be possible to hide this _hal_ticker implementation from other nRF5 driver modules, as the flash driver? I.e. de-couple flash driver from (hal)ticker? I could see no immediate drawbacks for this. And it would be a step towards increased modularity...

Reply:
This would require a generic coexist role (without air activity) to be implemented in the BLE controller. The role should then have a parameterized activity pattern that would fit for example the desired Flash acccess. However, I see this as a future activity that must be well thought before going in this direction. Another solution would be to move the ticker into a separate module.

Reply:
The convention followed here is inline with how applications using kernel includes header files, example for ARM (below order among tons of other include):
- include/arch/arm/cortex_m/error.h
- include/kernel.h
or for ARC
- include/arch/arc/v2/error.h
- include/kernel.h

The hal/ticker.h contains SoC specific macros, where as ticker/ticker.h has SoC independent APIs.

This said, I need to rework ticker.c to decouple the 32 KHz units used in there.